plugins/woocommerce/includes/shortcodes/class-wc-shortcode-my-account.php

```diff
@@ -89,6 +89,19 @@ private static function my_account_add_notices() {
 			/* translators: %s: logout url */
 			wc_add_notice( sprintf( __( 'Are you sure you want to log out? <a href="%s">Confirm and log out</a>', 'woocommerce' ), wc_logout_url() ) );
 		}
+
+		if ( get_user_option( 'default_password_nag' ) && wc_is_current_account_menu_item( 'dashboard' ) ) {
```

Comment:
Do we want to show this only on the main page? After creating the account, on the order confirmation page it links off to three pages, including one to manage your password.

On this page, the nag is not shown, and the user is asked to enter their current password. It might be useful to show the nag here too.